tests/contest/contest/src/tests/poststart_fail/mod.rs

```diff
@@ -0,0 +1,167 @@
+                    "-c".to_string(),
+                    "true".to_string(),
+                ])
+                .cwd("/")
+                .build()
+                .unwrap(),
+        )
+        .hooks(
+            HooksBuilder::default()
+                .poststart(vec![
+                    HookBuilder::default()
+                        .path("/bin/sh")
+                        .args(vec![
+                            "sh".to_string(),
+                            "-c".to_string(),
+                            format!("echo 'hook_1 called' >> {host_output_file}"),
+                        ])
+                        .build()
+                        .expect("could not build hook"),
+                    HookBuilder::default()
+                        .path("/bin/sh")
+                        .args(vec![
+                            "sh".to_string(),
+                            "-c".to_string(),
+                            format!("echo 'hook_2 called' >> {host_output_file}; exit 1"),
+                        ])
+                        .build()
+                        .expect("could not build hook"),
+                    HookBuilder::default()
+                        .path("/bin/sh")
+                        .args(vec![
+                            "sh".to_string(),
+                            "-c".to_string(),
+                            format!("echo 'hook_3 called' >> {host_output_file}"),
+                        ])
+                        .build()
+                        .expect("could not build hook"),
+                ])
+                .build()
+                .expect("could not build hooks"),
+        )
+        .build()
+        .unwrap()
+}
+
+/// Tests that when a poststart hook fails, subsequent hooks are not executed.
+///
+/// Validates that the runtime stops executing remaining poststart hooks after one fails,
+/// and returns an error (exit code 1). This test creates 3 hooks where hook_2 fails,
+/// then verifies that hook_1 and hook_2 ran but hook_3 did not.
+fn get_test(test_name: &'static str) -> Test {
+    Test::new(
+        test_name,
+        Box::new(move || {
+            let id = generate_uuid().to_string();
+            let bundle = prepare_bundle().unwrap();
+
+            let host_output_file = get_output_file_path(&bundle);
+
+            let spec = get_spec(host_output_file.to_str().unwrap());
+            set_config(&bundle, &spec).unwrap();
+
+            let create_result =
+                create_container(&id, &bundle, &CreateOptions::default()).map(|mut cmd| cmd.wait());
+
+            let create_failed = match create_result {
+                Err(_) => true,
+                Ok(Ok(status)) if !status.success() => true,
+                _ => false,
+            };
+
+            // runc doesn't follow spec and runs hooks at the create stage
+            // https://github.com/opencontainers/runc/issues/4347
+            if create_failed && !is_runtime_runc() {
```

Comment:
If runc’s behavior is incorrect, I think it’s fine to skip the test altogether.

Please refer to the code below. (use ConditionalTest)
https://github.com/youki-dev/youki/blob/main/tests/contest/contest/src/tests/misc_props/misc_props_test.rs#L95